units/src/amount/unsigned.rs

```diff
@@ -101,8 +101,8 @@ impl Amount {
     ///
     /// The function errors if the argument multiplied by the number of sats
     /// per bitcoin overflows a `u64` type.
-    pub fn from_int_btc(btc: u64) -> Result<Amount, OutOfRangeError> {
-        match btc.checked_mul(100_000_000) {
+    pub fn from_int_btc<T: Into<u64>>(whole_bitcoin: T) -> Result<Amount, OutOfRangeError> {
```

Comment:
This already merged, although `bitcoin_count` I think is more clear, probably too late to discuss this though.